### ISSUE TITLE
[Performance] MIGraphX + ROCm EP fallback triggers odd error state 

### ISSUE THREAD

**Issue by TedThemistokleous** [microsoft/onnxruntime#14679]
### Describe the issue

Running parity tests using parity_huggingface_gpt_attention.py noticed initial errors between ORT and pytorch output. When modifying parity_utils.py it appears that when removing either provider from the aggregate list of both, each run works without error.

From parity_utilities.py
```   if use_gpu:
        if torch.version.cuda:
            execution_providers.append("CUDAExecutionProvider")
        elif torch.version.hip:
            execution_providers.append("MIGraphXExecutionProvider")
            execution_providers.append("ROCMExecutionProvider")
```

Removing one of the EP so that only one provider is used and no fallback outside of CPU EP is used seems to work.
I've also added disable_fallback() and seen no change in results. 

Turned on logging after capturing output from said tests. 

Below are the logs taken with log severity set to 0 (Verbose)

[hugging_face_gpt_migx.log](https://github.com/microsoft/onnxruntime/files/10737384/hugging_face_gpt_migx.log)
[hugging_face_gpt_migx_rocm_fallback.log](https://github.com/microsoft/onnxruntime/files/10737385/hugging_face_gpt_migx_rocm_fallback.log)
[hugging_face_gpt_rocm.log](https://github.com/microsoft/onnxruntime/files/10737386/hugging_face_gpt_rocm.log)

### To reproduce

Run the latest branch of test_parity_huggingface_gpt_attention.py. Observe  the output errors. 

### Urgency

Critical

### Platform

Linux

### OS Version

20.04

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

Main

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

MIGraphX

### Execution Provider Library Version

ROCm 5.4.2

### Is this a quantized model?

Unknown

**Comment by TedThemistokleous** [comment]
ping @ytaous @PeixuanZuo . This is coming up in our latest testing.

Is there some additional logic we need in the MIGraphX EP to hand off/fallback gracefully? Seems like in the MIGraphX runs, since we don't currently support the attention operator, we should fallback to the ROCm EP.

**Comment by cloudhan** [comment]
Is this PR related? https://github.com/microsoft/onnxruntime/pull/14675

**Comment by TedThemistokleous** [comment]
Not sure, Looks like this change was just added. I'll retest this

**Comment by TedThemistokleous** [comment]
Reran the tests for this. I don't see a change in behavior.

[huggingface_debug_both_eps.log](https://github.com/microsoft/onnxruntime/files/10745358/huggingface_debug_both_eps.log)
[huggingface_debug_migx_EP_only.log](https://github.com/microsoft/onnxruntime/files/10745359/huggingface_debug_migx_EP_only.log)
[huggingface_debug_rocm_EP_only.log](https://github.com/microsoft/onnxruntime/files/10745360/huggingface_debug_rocm_EP_only.log)

I also noticed in the list adds CPUExecutionProvider first in parity utilities rather than the appropriate EP.  Achieved the same result here but moved things around incase ordering had a difference. 

```
    execution_providers = ["CPUExecutionProvider"]

    if use_gpu:
        if torch.version.cuda:
            execution_providers.append("CUDAExecutionProvider")
        elif torch.version.hip:
            execution_providers.append("MIGraphXExecutionProvider")
            execution_providers.append("ROCMExecutionProvider")
```

```
    execution_providers = []

    if use_gpu:
        if torch.version.cuda:
            execution_providers.append("CUDAExecutionProvider")
        elif torch.version.hip:
            execution_providers.append("MIGraphXExecutionProvider")
            execution_providers.append("ROCMExecutionProvider")

    execution_providers.append("CPUExecutionProvider")
```

Does this warrant a change for parity_utils.py here though? I thought EPs were selected based on ordering of the provider list passed to InferenceSession?

**Comment by TedThemistokleous** [comment]
@cloudhan  I've add a PR here for this change. 

https://github.com/microsoft/onnxruntime/pull/14700

**Comment by cloudhan** [comment]
I think this should be fixed with #14700, close for now. Feel free to reopen if it persist. Thanks for the contribution :)
